include/unifex/task.hpp

```diff
@@ -183,71 +215,81 @@ struct _promise {
-        return unifex::await_transform(*this, (Value&&) value);
-      } else if constexpr (tag_invocable<tag_t<unifex::await_transform>, type&, Value>
-          || detail::_awaitable<Value>) {
-        // Either await_transform has been customized or Value is an awaitable. Either
-        // way, we can dispatch to the await_transform CPO, then insert a transition back
-        // to the correct execution context if necessary.
-        return transform_awaitable_(unifex::await_transform(*this, (Value&&) value));
+        // We are co_await-ing a unifex::task, which completes inline because of
+        // task scheduler affinity. We don't need an additional transition.
+        return unifex::await_transform(*this, (Value &&) value);
+      } else if constexpr (
+          tag_invocable<tag_t<unifex::await_transform>, type&, Value> ||
+          detail::_awaitable<Value>) {
+        // Either await_transform has been customized or Value is an awaitable.
+        // Either way, we can dispatch to the await_transform CPO, then insert a
+        // transition back to the correct execution context if necessary.
+        return transform_awaitable_(
+            unifex::await_transform(*this, (Value &&) value));
       } else if constexpr (unifex::sender<Value>) {
-        return transform_sender_((Value&&) value);
+        return transform_sender_((Value &&) value);
       } else {
-        // Otherwise, we don't know how to await this type. Just return it and let the
-        // compiler issue a diagnostic.
-        return (Value&&) value;
+        // Otherwise, we don't know how to await this type. Just return it and
+        // let the compiler issue a diagnostic.
+        return (Value &&) value;
       }
     }
 
     template <typename Awaitable>
     decltype(auto) transform_awaitable_(Awaitable&& awaitable) {
       using blocking_t = decltype(blocking(awaitable));
 
-      if constexpr (!same_as<blocking_kind, blocking_t> && (blocking_kind::always_inline == blocking_t{})) {
-        return Awaitable{(Awaitable&&) awaitable};
+      if constexpr (
+          !same_as<blocking_kind, blocking_t> &&
+          (blocking_kind::always_inline == blocking_t{})) {
+        return Awaitable{(Awaitable &&) awaitable};
       } else {
         return unifex::await_transform(
             *this,
-            finally(as_sender((Awaitable&&) awaitable), unstoppable(schedule(this->sched_))));
+            finally(
+                as_sender((Awaitable &&) awaitable),
+                unstoppable(schedule(this->sched_))));
       }
     }
 
     template <typename Sender>
     decltype(auto) transform_sender_(Sender&& sndr) {
-      if constexpr (blocking_kind::always_inline == sender_traits<remove_cvref_t<Sender>>::blocking) {
-        return unifex::await_transform(*this, (Sender&&) sndr);
+      if constexpr (sender_traits<
+                        remove_cvref_t<Sender>>::is_always_scheduler_affine) {
```

Comment:
It was a mistake to merge this in this PR; *Scheduler* affinity is a weaker constraint than `always_inline` so there could be situations where we wake up from awaiting a *Sender* on the wrong *Scheduler*.  This will be addressed in #527.